bokehjs/src/coffee/common/embed.coffee

```diff
@@ -8,6 +8,18 @@ HasProperties = require "./has_properties"
 {pull_session} = require "./client"
 {Promise} = require "es6-promise"
 
+_handle_notebook_comms = (msg) ->
+  logger.debug("handling notebook comms")
+  doc = _.values(@)[0] # should only be one
```

Comment:
I know it's always one now, just trying to reduce the number of ways the code works.
I was imagining maybe in notebook_div do `render_items[0]['notebooktarget'] = comms_target` after calling `_standalone_docs_json_and_render_items`, and then embed.coffee would handle that case in the loop over the items. then you can pass each _handle_notebook_comms the single specific document and div that it will care about.
If you allow PUSH-DOC this may be especially useful because you'll need the div in order to replace the doc entirely.